### DISCUSSION TITLE
Variable layer height step size

### DISCUSSION THREAD

**Discussion by calania** [supermerill/SuperSlicer#2341] [Q&A]
Hello! I have used Cura forever but wanted to give superslicer a try.  I use variable layer height quite a lot and in Cura I was able to set what the Variable layer height step size should be. Basically I had 0.32 mm set as my default layer height but then it could use a layer height ±0.04. I read someone say that it was possible to this in superslicer to but I can't find a setting for it. Is this available and if so how do I do it?

**Comment by neophyl** [comment]
You can't configure discrete steps.  The minimum/maximum settings for the variable layer height function are determined by Printer Settings>Extruder 1>Layer Height Limits.

**Reply by supermerill** [reply]
If `z_step` is set to 0, yes. But if set, it constrains all layers to a multiple of it.

**Reply by neophyl** [reply]
Thaks Merill.  I wasn't aware of that.  I assumed those were for just setting up the proper values for your z stepping.  I have often wanted the variable layer height to be more granular.  Is there any problem setting those to be bigger values ?

**Reply by supermerill** [reply]
no. But if you set a value so big that you can't have one between the min and max, then It's undefined behavior.

**Comment by supermerill** [comment]
Printer settings -> Size and coordinates section -> z utilities line -> Z full step
(or search for `z_step`)

**Reply by calania** [reply]
Thanks and sorry for my late reply. I can't get it working but that it probably me just using it wrong. If I want the layer height to be ±0.04 what should I then set my z full step to?

**Reply by supermerill** [reply]
0.04
then, the layer height can only be a multiple of 0.04, like 1.24
